.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+base_url="https://services.nvd.nist.gov/rest/json/cves/2.0"
+results_per_page=2000
+dir_name="nvd-data"
+
+# Function to fetch data for a given period
+fetch_data_for_period () {
+  local period_start_date=$1
+  local period_end_date=$2
+  local current_start_index=0
+
+  while : ; do
+    local file_name="${period_start_date}_${current_start_index}.json"
+
+    echo "Fetching data from $period_start_date to $period_end_date, starting at index $current_start_index..."
+
+    retry_intervals=(6 12 24) # Array of retry intervals in seconds
+    success=false
+
+    for i in "${retry_intervals[@]}"; do
+        if curl -s --fail --show-error "$base_url?pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
```

Comment:
Since we have braced and non-braced string interpolation in this script, I might spring to use the braced version here to make the full URL easier to read.